src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs

```diff
@@ -351,8 +343,39 @@ void PostProcessEscapedArgument()
                         var pointsToValue = GetPointsToAbstractValue(operation.Value);
                         HandlePossibleEscapingOperation(operation, pointsToValue.Locations);
                     }
+                    else if (FlowBranchConditionKind == ControlFlowConditionKind.WhenFalse &&
+                        operation.Parameter.RefKind == RefKind.Out &&
+                        operation.Parent is IInvocationOperation invocation &&
+                        invocation.TargetMethod.ReturnType.SpecialType == SpecialType.System_Boolean)
+                    {
```

Comment:
Can't think of anything of the top of my head, but this seems like a heuristic to keep an eye on in the future.  Otherwise LGTM.